protocol_rfcs/variant-type.md

```diff
@@ -0,0 +1,87 @@
+# Variant Data Type
+**Associated Github issue for discussions: https://github.com/delta-io/delta/issues/2864**
+
+This protocol change adds support for the Variant data type.
+The Variant data type is beneficial for storing and processing semi-structure data.
+
+--------
+
+> ***New Section after the [Clustered Table](#clustered-table) section***
```

Comment:
Would be better to follow the structure of other features in the protocol doc - Requirements for readers and requirements for writers.

Reply:
Updated with reader and writer requirements.